### PR TITLE
[video_player] Update Pigeon for non-nullable generics

### PR DESCRIPTION
**Pull request by stuartmorgan**
Updates the Pigeon definition to uses non-nullable generics now that it's possible.

In practice for this plugin it's currently a no-op: the only affected collection is passed from Dart to native, on the Dart side the source was already a `Map<String, String>`, and on the native side we are using languages without nullability support in generics (Java and Obj-C). However, doing this now eliminates tech debt, and will avoid problems if/when the native side switches to Kotlin/Swift.

On iOS, also removes a class that was accidentally left, unused, in the Pigeon definition file after an earlier update to that file. Current versions of Pigeon would generate that class even though it's not used, so removing it from the definition avoids changes to the generated code that would otherwise have been caused by the update.

Part of https://github.com/flutter/flutter/issues/155891

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/video_player/video_player_android/pubspec.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
-  pigeon: ^9.2.5
```
Comment (matanlurey) [inline packages/video_player/video_player_android/pubspec.yaml:28]: Wow

**Review by matanlurey — APPROVED** [review]

File: packages/video_player/video_player_avfoundation/pigeons/messages.dart [hunk 2]
```diff
@@ -23,12 +23,7 @@ class CreationOptions {
   String? uri;
   String? packageName;
   String? formatHint;
-  Map<String?, String?> httpHeaders;
-}
-
-class MixWithOthersMessage {
-  MixWithOthersMessage(this.mixWithOthers);
-  bool mixWithOthers;
```
Comment (tarrinneal) [inline packages/video_player/video_player_avfoundation/pigeons/messages.dart:31]: Was this just unused before?

**Review by tarrinneal — APPROVED** [review]
Seems like another plug and play. :)

File: packages/video_player/video_player_avfoundation/pigeons/messages.dart (hunk 2 above)
Comment (stuartmorgan) [reply packages/video_player/video_player_avfoundation/pigeons/messages.dart:31]: Yes, I somehow missed it in https://github.com/flutter/packages/pull/6507

**Pull request merged**